core/vm/instructions.go

```diff
@@ -820,18 +820,17 @@ func makePush(size uint64, pushByteSize int) executionFunc {
 	return func(pc *uint64, interpreter *EVMInterpreter, callContext *callCtx) ([]byte, error) {
 		codeLen := len(callContext.contract.Code)
 
-		startMin := codeLen
-		if int(*pc+1) < startMin {
-			startMin = int(*pc + 1)
+		startMin := int(*pc + 1)
+		if startMin >= codeLen {
+			startMin = codeLen
 		}
-
-		endMin := codeLen
-		if startMin+pushByteSize < endMin {
-			endMin = startMin + pushByteSize
+		endMin := startMin + pushByteSize
+		if startMin+pushByteSize >= codeLen {
+			endMin = codeLen
 		}
 
 		integer := new(uint256.Int)
-		callContext.stack.push(integer.SetBytes(common.RightPadBytes(callContext.contract.Code[startMin:endMin], pushByteSize)))
+		callContext.stack.push(common.SetBytesRightPadded(integer, callContext.contract.Code[startMin:endMin], pushByteSize))
 
```

Comment:
I think we don't need to pad bytes at all: integer.SetBytes does that automatically. The line can be simplified to
callContext.stack.push(integer.SetBytes(callContext.contract.Code[startMin:endMin]))

Reply:
Potentially `common.RightPadBytes` call there was not really required

Reply:
A nice idea. I tried to add this to `TestRightPadding`:
```
    TestRightPadding: big_test.go:28: not equal n=1 l=2 slice=[1]
        Got		[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0]
        !=
        Expect	[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1]((*uint256.Int)(0xc0000189c0)((len=4 cap=4) {
         (uint64) 1,
         (uint64) 0,
         (uint64) 0,
         (uint64) 0
        })
        )
```

I think the padding is essential here

Reply:
Sorry, my bad. We only need to right-pad if endMin-startMin != pushByteSize, which only happens if we're overflowing codeLen.

Reply:
A good idea, I'll and the check.

Reply:
But then, on the other hand, your method SetBytesRightPadded has little overhead compared to SetBytes, and we can call it all the time regardless.

Reply:
Actually, there's no need to pad, at all. It's possible to cheat here: if we're at the end of the code, the execution will stop. It doesn't matter what's in the stack, so one could just as well just push an empty value.

Reply:
Thank you Martin for the insight, I've created a PR to simplify: https://github.com/ledgerwatch/turbo-geth/pull/638